natiami/assignment3/FindWordsTest.cpp

```diff
@@ -0,0 +1,51 @@
+#include "FindWordsTest.h"
+
+void FindWordsTest::test(){
+	test_findAllWords();
+	test_EmptyGrid();
+	test_EmptyDictionary();
+}
+
+Grid* FindWordsTest::initGrid(){
+	string gridRow1 = "AAR";
+	string gridRow2 = "TCD";
+	Grid* grid = new Grid(3, 2);
+	for (size_t j = 0; j < grid->width; j++)
+	{
+		grid->cells[0][j] = gridRow1[j];
+		grid->cells[1][j] = gridRow2[j];
+	}
+	return grid;
+}
+
+Dictionary* FindWordsTest::initDictionary(){
+	set<string> words = { "CAR", "CARD", "CART", "CAT" };
+	set<string> prefixes = { "C", "CA", "CAR", "CARD", "CART", "CAT" };
+	Dictionary* dictionary = new Dictionary(words, prefixes);
+	return dictionary;
+}
+
+void FindWordsTest::test_findAllWords(){
+	Dictionary* dictionary = initDictionary();
+	Grid* grid = initGrid();
+	set<string> found_words = grid->findAllWords(dictionary);
+	set<string> expected_words = { "CAR", "CARD", "CAT" };
+	EXPECT_EQ(expected_words, found_words);
+}
+
+void FindWordsTest::test_EmptyGrid(){
+	Dictionary* dictionary = initDictionary();
+	Grid* grid = new Grid();

```

Comment:
Tip:
You can initialize the Grid by value, and give it a meaningful name:
```
Grid emptyGrid;
set<string> empty_expected_words;
EXPECT_EQ(empty_expected_words, empty_grid->findAllWords(dictionary));
```
This way this test is read like a sentence:
"Expect an empty result if searching the empty grid".

Also there is would be no memory leak of the grid.